tests/functional/sources/test_source_configs.py

```diff
@@ -0,0 +1,594 @@
+import pytest
+from dbt.contracts.graph.model_config import SourceConfig
+from dbt.contracts.graph.unparsed import FreshnessThreshold, Time, TimePeriod, Quoting
+from dbt.exceptions import CompilationException
+
+from dbt.tests.util import run_dbt, update_config_file, get_manifest
+
+
+class SourceConfigTests:
+    @pytest.fixture(scope="class", autouse=True)
+    def setUp(self):
+        pytest.expected_config = SourceConfig(
+            enabled=True,
+            # TODO: uncomment all this once it's added to SourceConfig, throws error right now
+            # quoting = Quoting(database=False, schema=False, identifier=False, column=False)
+            # freshness = FreshnessThreshold(
+            #     warn_after=Time(count=12, period=TimePeriod.hour),
+            #     error_after=Time(count=24, period=TimePeriod.hour),
+            #     filter=None
+            #     )
+            # loader = "a_loader"
+            # loaded_at_field = some_column
+            # database = custom_database
+            # schema = custom_schema
+            # # identifier = "seed"  #this doesnt seems to be supported now?
+            # meta = {'languages': ['python']}
+            # tags = ["important_tag"]
+        )
+
+
+models__schema_yml = """version: 2
+
+sources:
+  - name: test_source
+    tables:
+      - name: test_table
+  - name: other_source
+    tables:
+      - name: test_table
+"""
+
+
+# Test enabled config in dbt_project.yml
+# expect pass, already implemented
+class TestSourceEnabledConfigProjectLevel(SourceConfigTests):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "schema.yml": models__schema_yml,
+        }
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "sources": {
+                "test": {
+                    "test_source": {
+                        "enabled": True,
+                    },
+                }
+            }
+        }
+
+    def test_enabled_source_config_dbt_project(self, project):
+        run_dbt(["compile"])
```

Comment:
`compile` gets the job done; I'm also thinking this could just be `parse`, since we _just_ need the output of parsing. That's one way to draw the interface boundary, and denote this as Language behavior.

Do we want to develop consistent guidance around which commands we use, as a proxy for the functionality they're testing? Or are there other better ways to decorate tests, to denote their surface area?

Reply:
That makes a lot of sense.  `parse` it is!

I think the legacy (can I call them that yet?) tests use `run` way more often than is strictly needed.  I'm not sure what guidance we would develop short of thinking through what we really are testing.

We do have the option with pytest to create [custom markers](https://docs.pytest.org/en/6.2.x/example/markers.html) to mark what team a test applies to.  But I think it's important to define why we need to do this and what value it would add.